shared/params.go

```diff
@@ -58,17 +59,100 @@ func ParseSHAParamFull(r *http.Request) (runSHA string, err error) {
 	return runSHA, err
 }
 
+// ParseProductAtRevision parses a test-run spec into a ProductAtRevision struct.
+func ParseProductAtRevision(spec string) (productAtRevision ProductAtRevision, err error) {
+	pieces := strings.Split(spec, "@")
+	if len(pieces) > 2 {
+		return productAtRevision, errors.New("invalid product@revision spec: " + spec)
+	}
+	if productAtRevision.Product, err = ParseProduct(pieces[0]); err != nil {
+		return productAtRevision, err
+	}
+	if len(pieces) < 2 {
+		// No @ is assumed to be the product only.
+		productAtRevision.Revision = "latest"
+	} else {
+		productAtRevision.Revision = pieces[1]
+	}
+	return productAtRevision, nil
+}
+
+// ParseProduct parses the `browser-version-os-version` input as a Product struct.
+func ParseProduct(product string) (result Product, err error) {
+	pieces := strings.Split(product, "-")
+	if len(pieces) > 4 {
+		return result, fmt.Errorf("invalid product: %s", product)
+	}
+	result = Product{
+		BrowserName: pieces[0],
+	}
+	if !IsBrowserName(result.BrowserName) {
+		return result, fmt.Errorf("invalid browser name: %s", result.BrowserName)
+	}
+	if len(pieces) > 1 {
+		if _, err := ParseVersion(pieces[1]); err != nil {
+			return result, fmt.Errorf("invalid browser version: %s", pieces[1])
```

Comment:
I considered that (version optional and digit regex for disambiguation), and actually started coding it, but decided on deferring until later, so that we can be thoughtful (as you suggest here).

Reply:
OK, let's see about it later :)